File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.cpp
class Solution {
public:
    int missingInteger(vector<int>& nums) {
        int res = nums[0];
        int n = nums.size();
        unordered_set<int> found(nums.begin(), nums.end());
        int idx = 1;
        while(idx < n && nums[idx] == nums[idx - 1] + 1)
        {
            res += nums[idx];
            idx++;
        }
        while(found.count(res))
        {
            res++;
        }
        return res;
    }
};

File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.py
class Solution:
    def missingInteger(self, nums: List[int]) -> int:
        res = nums[0]
        n = len(nums)
        found = set(nums)
        idx = 1
        while idx < n and nums[idx] == nums[idx - 1] + 1:
            res += nums[idx]
            idx += 1
        while res in found:
            res += 1
        return res
